example/spec/parallel/startSF/startSfSpec.js

```diff
@@ -0,0 +1,113 @@
+const { loadConfig, createTimestampedTestId, timestampedName } = require('../../helpers/testUtils');
+
+const config = loadConfig();
+
+const testName = createTimestampedTestId(config.stackName, 'testStartSf');
+
+const passSfName = timestampedName('passTestSf');
+const passSfDef = {
+  Comment: 'Pass-only step function',
+  StartAt: 'PassState',
+  States: {
+    PassState: {
+      Type: 'Pass',
+      Result: '$.payload',
+      End: true
+    }
+  }
+};
+const passSfRoleArn = `arn:aws:iam::${config.awsAccountId}:role/${config.stackName}-steprole`;
+
+const passSfParams = {
+  name: passSfName,
+  definition: JSON.stringify(passSfDef),
+  roleArn: passSfRoleArn
+};
+
+const sfStarterName = `${config.stackName}-sqs2sf`;
+
+function generateStartSfMessages(num, workflowArn) {
+  const arr = [];
+  for (let i = 0; i < num; i += 1) {
+    arr.push({ cumulus_meta: { state_machine: workflowArn }, payload: `message #${i}` });
+  }
+  return arr;
+}
+
+describe('the sf-starter lambda function', () => {
+  let queueUrl;
+
+  beforeAll(async () => {
+    const { QueueUrl } = await sqs().createQueue({
+      QueueName: `${testName}Queue`,
+      Attributes: {
+        VisibilityTimeout: '0'
+      }
+    }).promise();
+    queueUrl = QueueUrl;
+  });
+
+  afterAll(async () => {
+    await sqs().deleteQueue({
+      QueueUrl: queueUrl
+    }).promise();
+  });
+
+  it('has a configurable message limit', () => {
+    pending('CW API does not return rule inputs. Pass confirmed in manual testing.');
+    const messageLimit = config.sqs_consumer_rate; // not available in config
+    expect(messageLimit).toBeDefined();
+  });
+
+  describe('when provided a queue', () => {
+    let passSfArn;
+    let qAttrParams;
+    let messagesConsumed;
+
+    beforeAll(async () => {
+      qAttrParams = {
+        QueueUrl: queueUrl,
+        AttributeNames: ['ApproximateNumberOfMessages', 'ApproximateNumberOfMessagesNotVisible']
+      };
+      const { stateMachineArn } = await sfn().createStateMachine(passSfParams).promise();
+      passSfArn = stateMachineArn;
+      const msgs = generateStartSfMessages(30, passSfArn);
+      await Promise.all(msgs.map((msg) => sqs().sendMessage({ QueueUrl: queueUrl, MessageBody: JSON.stringify(msg) }).promise()));
+    });
+
+    afterAll(async () => {
+      await sfn().deleteStateMachine({ stateMachineArn: passSfArn }).promise();
+    });
+
+    it('that has messages', async () => {
+      const { Attributes } = await sqs().getQueueAttributes(qAttrParams).promise();
+      expect(Attributes.ApproximateNumberOfMessages).toBe('30');
+    });
+
+    it('consumes the messages', async () => {
+      const { Payload } = await lambda().invoke({
+        FunctionName: sfStarterName,
+        InvocationType: 'RequestResponse',
+        Payload: JSON.stringify({
+          queueUrl,
+          messageLimit: 25
+        })
+      }).promise();
+      messagesConsumed = parseInt(Payload, 10);
+      expect(messagesConsumed).toBeGreaterThan(0);
```

Comment:
So another idea would be to drop the messageLimit in the Payload to something >1 but still less than what's reasonably expected to be processed in onestep 4 or 5.  

Then we can assert that we are multiprocessing, but also limiting.  I'm not sure I'm kind of thinking out loud here.  so you could expect(messagesConsumed).toBeEqual(5) for example.  It's kinda race-conditiony though since SQS doesn't guarantee how many messages it give you.  So maybe this isn't a great idea.

Reply:
I agree with your first comment. As you noted, SQS doesn't guarantee the number of messages returned, which directly affects the number of executions started so it's not really possible to test for a fixed number.